engine.py

```diff
@@ -882,12 +949,136 @@ def _get_dialog_parent(self):
                 # try to get a hold of the main window. it seems to be the only
                 # one with windowIconText set. There should be a better way to
                 # do this.
-                if (widget.isWindow() and
-                        not isinstance(widget, QtGui.QDialog) and
-                        widget.windowIconText()):
+                if (
+                    widget.isWindow()
+                    and not isinstance(widget, QtGui.QDialog)
+                    and widget.windowIconText()
+                ):
                     parent = widget
 
-        self.logger.debug(
-            "Found top level widget %s for dialog parenting" % (parent,))
+        self.logger.debug("Found top level widget %s for dialog parenting" % (parent,))
         return parent
 
+    def node_handler(self, node):
+        """
+        Get the node handler hook for the given node.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :returns: A :class:`NodeHandlerBase` instance.
+        """
+        if not node:
+            return
+        node_type = node.type()
+        node_type_name = node_type.name()
+        node_category = node_type.category().name()
+        hook_instance = self.__node_handlers.get(node_category, {}).get(
+            node_type_name, False
+        )
+
+        if hook_instance is False:
+            if not self.__node_handlers.get(node_category):
+                self.__node_handlers[node_category] = {}
```

Comment:
`setdefault` will return current value for `node_category` after setting it to `{}` if no `node_category` exists

```suggestion
        category_handler = self.__node_handlers.setdefault(node_category, {})
        hook_instance = category_handler.get(node_type_name, False)

        if hook_instance is False:
```

Or if it works better, switch `self.__node_handlers` to be constructed with `defaultdict(dict)` with this at the top of the file `from collections import defaultdict`